### ISSUE TITLE
Problem with extra coordinate on 32-bit system

### ISSUE THREAD

**Issue by ViviCoder** [sunpy/ndcube#133]
While running tests on a 32-bit system, I got the following errors (this is with version 1.0.1, but I obtain similar results with the master version):
```
============================= test session starts ==============================
platform linux -- Python 3.7.1, pytest-3.6.4, py-1.7.0, pluggy-0.8.0
rootdir: /build/1st/ndcube-1.0.1, inifile: setup.cfg
plugins: remotedata-0.3.1, openfiles-0.3.0, doctestplus-0.1.3, arraydiff-0.2
collected 95 items / 4 errors

==================================== ERRORS ====================================
_________________ ERROR collecting ndcube/tests/test_ndcube.py _________________
../../../ndcube/tests/test_ndcube.py:123: in <module>
    [(cubem[:, 1],
../../../ndcube/mixins/ndslicing.py:31: in __getitem__
    return super().__getitem__(item)
/usr/lib/python3/dist-packages/astropy/nddata/mixins/ndslicing.py:61: in __getitem__
    return self.__class__(**kwargs)
../../../ndcube/ndcube.py:185: in __init__
    extra_coords, self.missing_axis, data.shape)
../../../ndcube/utils/cube.py:90: in _format_input_extra_coords_to_extra_coords_wcs_axis
    raise ValueError(coord_1_format_error)
E   ValueError: 2nd element of extra coordinate tuple must be None or an int giving the data axis to which the coordinate corresponds.
------------------------------- Captured stdout --------------------------------
[{'coordinate_type': None, 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'spectral', 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 1}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 0}]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
------------------------------- Captured stderr --------------------------------
WARNING: FITSFixedWarning: 'celfix' made the change 'Unmatched celestial axes'. [astropy.wcs.wcs]
/build/1st/ndcube-1.0.1/ndcube/tests/test_ndcube.py:65: RuntimeWarning: invalid value encountered in sqrt
  uncertainty = np.sqrt(data)
_________________ ERROR collecting ndcube/tests/test_ndcube.py _________________
../../../ndcube/tests/test_ndcube.py:123: in <module>
    [(cubem[:, 1],
../../../ndcube/mixins/ndslicing.py:31: in __getitem__
    return super().__getitem__(item)
/usr/lib/python3/dist-packages/astropy/nddata/mixins/ndslicing.py:61: in __getitem__
    return self.__class__(**kwargs)
../../../ndcube/ndcube.py:185: in __init__
    extra_coords, self.missing_axis, data.shape)
../../../ndcube/utils/cube.py:90: in _format_input_extra_coords_to_extra_coords_wcs_axis
    raise ValueError(coord_1_format_error)
E   ValueError: 2nd element of extra coordinate tuple must be None or an int giving the data axis to which the coordinate corresponds.
------------------------------- Captured stdout --------------------------------
[{'coordinate_type': None, 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'spectral', 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 1}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 0}]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
------------------------------- Captured stderr --------------------------------
/build/1st/ndcube-1.0.1/ndcube/tests/test_ndcube.py:65: RuntimeWarning: invalid value encountered in sqrt
  uncertainty = np.sqrt(data)
_____________ ERROR collecting ndcube/tests/test_ndcubesequence.py _____________
../../../ndcube/tests/test_ndcubesequence.py:115: in <module>
    (seq[0], NDCube),
../../../ndcube/ndcube_sequence.py:84: in __getitem__
    return utils.sequence.slice_sequence(self, item)
../../../ndcube/utils/sequence.py:57: in slice_sequence
    return slice_sequence_by_sequence_items(cubesequence, sequence_items)
../../../ndcube/utils/sequence.py:211: in slice_sequence_by_sequence_items
    return result.data[sequence_items[0].sequence_index][sequence_items[0].cube_item]
../../../ndcube/mixins/ndslicing.py:31: in __getitem__
    return super().__getitem__(item)
/usr/lib/python3/dist-packages/astropy/nddata/mixins/ndslicing.py:61: in __getitem__
    return self.__class__(**kwargs)
../../../ndcube/ndcube.py:185: in __init__
    extra_coords, self.missing_axis, data.shape)
../../../ndcube/utils/cube.py:90: in _format_input_extra_coords_to_extra_coords_wcs_axis
    raise ValueError(coord_1_format_error)
E   ValueError: 2nd element of extra coordinate tuple must be None or an int giving the data axis to which the coordinate corresponds.
------------------------------- Captured stdout --------------------------------
[{'coordinate_type': None, 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'spectral', 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 1}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 0}]
_____________ ERROR collecting ndcube/tests/test_ndcubesequence.py _____________
../../../ndcube/tests/test_ndcubesequence.py:115: in <module>
    (seq[0], NDCube),
../../../ndcube/ndcube_sequence.py:84: in __getitem__
    return utils.sequence.slice_sequence(self, item)
../../../ndcube/utils/sequence.py:57: in slice_sequence
    return slice_sequence_by_sequence_items(cubesequence, sequence_items)
../../../ndcube/utils/sequence.py:211: in slice_sequence_by_sequence_items
    return result.data[sequence_items[0].sequence_index][sequence_items[0].cube_item]
../../../ndcube/mixins/ndslicing.py:31: in __getitem__
    return super().__getitem__(item)
/usr/lib/python3/dist-packages/astropy/nddata/mixins/ndslicing.py:61: in __getitem__
    return self.__class__(**kwargs)
../../../ndcube/ndcube.py:185: in __init__
    extra_coords, self.missing_axis, data.shape)
../../../ndcube/utils/cube.py:90: in _format_input_extra_coords_to_extra_coords_wcs_axis
    raise ValueError(coord_1_format_error)
E   ValueError: 2nd element of extra coordinate tuple must be None or an int giving the data axis to which the coordinate corresponds.
------------------------------- Captured stdout --------------------------------
[{'coordinate_type': None, 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'spectral', 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 1}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 0}]
!!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!
```

**Comment by DanRyanIrish** [comment]
Hi @ViviCoder.  Thanks so much for posting this issue.  We'll take a look at it.

**Comment by ViviCoder** [comment]
@DanRyanIrish Is this issue fixed in master? I am asking because it was closed before 1.1.3 was released, but it seems to be still present in 1.1.3.

**Comment by DanRyanIrish** [comment]
@ViviCoder, this should be resolved in 1.3.2.
